Simplify ResourceOriginAuthorizationInterceptor by consolidating deleted logic

rolandgroen opened the pull request:
Moved from catching an exception to using the overloaded method with logic around te delete flag.

rolandgroen committed:
Simplify ResourceOriginAuthorizationInterceptor by consolidating deleted resource checks and streamlining exception handling

rolandgroen committed:
Ensure all remote branches are fetched before mirroring to GitLab

rolandgroen committed:
Update GitLab mirror workflow to refine branch and tag synchronization

Revised branch fetching to use remote tracking branches and adjusted push logic to prevent accidental deletions while maintaining force synchronization. Simplified tag push command for clarity.

JorisHeadease committed:
KPTSTD-912: Fixing unit tests as the dao is called with 3 params and mockito was missing

JorisHeadease commented on .github/workflows/gitlab-mirror.yaml:
```diff
@@ -40,6 +40,11 @@ jobs:
+          git push gitlab 'refs/remotes/origin/*:refs/heads/*' --force
+
+          # 4. push all tags
+          git push gitlab --tags --force
```
These changes can cause the complete sync to stop working when someone pushes code to GitLab right?

JorisHeadease reviewed (commented):
Thanks @rolandgroen, much simpler like this indeed. I fixed the unit tests, and I'm not sure if we want the GitLab mirror changes like this

rolandgroen committed:
Refine GitLab mirror workflow for better branch and tag synchronization

Updated branch fetching to use `--all` with `--prune` for cleaner synchronization. Adjusted push logic to mirror branches and tags with `--prune` and `--force`, ensuring accurate synchronization and avoiding inconsistencies.

rolandgroen committed:
Refine GitLab mirror workflow to fetch tags instead of all repos for improved synchronization

rolandgroen committed:
Reverting back to the old situation, doesn't need to be fixed here
